### PR TITLE
support trusty and xenial as osquery repository allows, xenial seems …

### PR DESCRIPTION
**Pull request by juju4**
…to require apt-transport-https

### CODE REVIEW

File: tasks/Ubuntu.yml [hunk 1]
```diff
@@ -2,24 +2,29 @@
 # tasks file for osquery (Ubuntu specific)
 
 - name: Download osquery APT key
-  become: yes
```
Comment (kbrebanov) [inline tasks/Ubuntu.yml:5]: Why are you removing the 'become' directive?

File: tasks/Ubuntu.yml [hunk 2]
```diff
@@ -2,24 +2,29 @@
...
+- name: Ensure apt-transport-https is installed
+  apt:
+    name: apt-transport-https
+    state: present
+  when: ansible_distribution_release == 'xenial'
+  tags:
+    - osquery
+
 - name: Configure osquery APT repository
-  become: yes
```
Comment (kbrebanov) [inline tasks/Ubuntu.yml:14]: Why are you removing the 'become' directive?

File: tasks/Ubuntu.yml [hunk 3]
```diff
@@ -2,24 +2,29 @@
...
   apt_repository:
-    repo: 'deb [arch=amd64] https://osquery-packages.s3.amazonaws.com/trusty trusty main'
+    repo: 'deb [arch=amd64] https://osquery-packages.s3.amazonaws.com/{{ ansible_distribution_release }} {{ ansible_distribution_release }} main'
     state: present
   tags:
     - osquery
 
 - name: Install osquery packages
-  become: yes
```
Comment (kbrebanov) [inline tasks/Ubuntu.yml:22]: Why are you removing the 'become' directive?

File: tasks/Ubuntu.yml [hunk 4]
```diff
@@ -2,24 +2,29 @@
...
     id: C9D8B80B
     state: present
   tags:
     - osquery
 
+- name: Ensure apt-transport-https is installed
+  apt:
+    name: apt-transport-https
+    state: present
+  when: ansible_distribution_release == 'xenial'
```
Comment (kbrebanov) [inline tasks/Ubuntu.yml]: No need to have a conditional checking if it's Xenial as it's ok to also ensure it's installed on Trusty.

File: tasks/Ubuntu.yml [hunk 5]
```diff
@@ -2,24 +2,29 @@
...
+  tags:
+    - osquery
+
 - name: Configure osquery APT repository
-  become: yes
   apt_repository:
-    repo: 'deb [arch=amd64] https://osquery-packages.s3.amazonaws.com/trusty trusty main'
+    repo: 'deb [arch=amd64] https://osquery-packages.s3.amazonaws.com/{{ ansible_distribution_release }} {{ ansible_distribution_release }} main'
```
Comment (kbrebanov) [inline tasks/Ubuntu.yml]: I would recommend using the 'lower' filter.

`ansible_distribution_release|lower`

File: tasks/Ubuntu.yml [hunk 6]
```diff
@@ -28,3 +33,4 @@
     - "{{ osquery_packages }}"
   tags:
     - osquery
+
```
Comment (kbrebanov) [inline tasks/Ubuntu.yml]: Can you remove the trailing newline?

File: tasks/main.yml [hunk 7]
```diff
@@ -15,3 +15,4 @@
   when: ansible_distribution == "Ubuntu"
   tags:
     - osquery
+
```
Comment (kbrebanov) [inline tasks/main.yml]: Can you remove the trailing newline?

**Review by kbrebanov — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 7 above)
Comment (kbrebanov) [reply tasks/main.yml]: Or better yet, remove the file from the commit since nothing really needs to be changed in it.

**Review by kbrebanov — APPROVED** [review]

**Comment by juju4** [comment]
Thanks Kevin!

**Pull request merged**